### DISCUSSION TITLE
calculate wind effect on UAV using JSBSIM

### DISCUSSION THREAD

**Discussion by sorgan191** [JSBSim-Team/jsbsim#731] [Support requests]
I am working on generate path planning for UAVs, I have a data set containing wind direction and speed. I want to know how much the wind will shift the drone from its path (meters) using JSBSIM.
Note that I am a very new user of JSBSIM.

thank you in advance

**Comment by seanmcleod** [comment]
So are you saying that the drone won't adjust it's heading to follow the pre-planned route (course) when it detects via GPS etc. that it is drifting off-route?

Here is an example of the sort of heading change required given the wind vector, the aircraft's airspeed in order to fly the required course.

![image](https://user-images.githubusercontent.com/824856/191000019-d9edf690-f104-4355-9a63-2da5dc5501c2.png)

If the UAV isn't going to change it's heading then it will drift over time.

![image](https://user-images.githubusercontent.com/824856/191000888-042c772d-91b5-46b5-86ff-73c740c0bc47.png)

The calculation is basic vector maths.

**Comment by sorgan191** [comment]
Good evening,

First of all thanks for the instant response, then yes I want to know what
the new location (x,y,z) and acceleration on each axis is.

best regards

**Comment by seanmcleod** [comment]
So, you want to fly the UAV in JSBSim following some heading commands etc. and input a wind vector and then you want to track the 3D position and the acceleration in each axis?

If so, that's fairly easy to do in JSBSim since there are logging facilities, e.g. see the `<output>` element that will allow you to log lots of different state information including 3D position, accelerations etc.

**Comment by bcoconni** [comment]
This topic is neither a bug nor a feature request.
This is transferred to "Discussions".
